src/geo.c

```diff
@@ -554,9 +558,12 @@ void georadiusGeneric(client *c, int srcKeyIndex, int flags) {
             } else if (!strcasecmp(arg, "count") && (i+1) < remaining) {
                 if (getLongLongFromObjectOrReply(c, c->argv[base_args+i+1],
                     &count, NULL) != C_OK) return;
-                if (count <= 0) {
-                    addReplyError(c,"COUNT must be > 0");
+                if (count == 0) {
+                    addReplyError(c,"COUNT must be greater than 0 or less than 0");
                     return;
+                } else if (count < 0) {
```

Comment:
Whether to allow count < 0?

Reply:
`count < 0` is the new syntax, which means fast return, refer: #8245

Reply:
```c
long returned_items = (count == 0 || result_length < count) ?
                          result_length : count;
```

`count == 0` Does it also need to be update.